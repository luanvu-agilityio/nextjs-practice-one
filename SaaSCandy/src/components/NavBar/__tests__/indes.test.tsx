import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import Navbar from '../index';

jest.mock('next/link', () => {
  const MockedLink = ({
    children,
    href,
  }: {
    children: React.ReactNode;
    href: string;
  }) => {
    return <a href={href}>{children}</a>;
  };
  MockedLink.displayName = 'MockedLink';
  return MockedLink;
});

// Mock constants
jest.mock('@/constants', () => ({
  NAV_LINKS: [
    { href: '/home', label: 'Home' },
    { href: '/about', label: 'About' },
    { href: '/contact', label: 'Contact' },
  ],
}));

describe('Navbar - Snapshot', () => {
  it('matches snapshot', () => {
    const { container } = render(<Navbar />);
    expect(container).toMatchSnapshot();
  });
});

describe('Navbar - Interactive', () => {
  it('toggles mobile menu on button click', async () => {
    const user = userEvent.setup();
    render(<Navbar />);

    const toggleButton = screen.getByRole('button', { name: /toggle menu/i });

    // Click to open menu
    await user.click(toggleButton);

    // Menu links should now be visible
    const homeLinks = screen.getAllByText('Home');
    expect(homeLinks.length).toBeGreaterThan(0);
    expect(screen.getAllByText('About')).toHaveLength(2);

    // Click to close menu
    await user.click(toggleButton);

    // Only desktop link remains (hidden on mobile with CSS)
    expect(screen.getAllByText('Home')).toHaveLength(1);
  });
});
