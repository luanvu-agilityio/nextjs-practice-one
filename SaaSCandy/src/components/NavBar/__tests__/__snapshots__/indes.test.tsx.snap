// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Navbar - Snapshot matches snapshot 1`] = `
<div>
  <nav
    class="flex-1"
  >
    <ul
      class="hidden md:flex items-center justify-start gap-10 text-primary w-full"
    >
      <li>
        <a
          href="/home"
        >
          Home
        </a>
      </li>
      <li>
        <a
          href="/about"
        >
          About
        </a>
      </li>
      <li>
        <a
          href="/contact"
        >
          Contact
        </a>
      </li>
    </ul>
    <div
      class="md:hidden flex items-center"
    >
      <button
        aria-label="Toggle menu"
        class="cursor-pointer inline-flex items-center justify-center rounded-full transition-colors focus-visible:outline-none focus-visible:ring-0 focus-visible:ring-ring focus-visible:ring-offset-0 disabled:opacity-50 disabled:cursor-not-allowed button-animation bg-transparent border-none text-primary hover:opacity-90 w-6 h-6 p-4"
        data-testid="icon-button"
      >
        <svg
          class="text-gray-700"
          fill="none"
          height="18"
          viewBox="0 0 24 24"
          width="18"
        >
          <path
            d="M3 12h18M3 6h18M3 18h18"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="1.6"
          />
        </svg>
      </button>
    </div>
  </nav>
</div>
`;
